system/jlib/jevent.cpp

```diff
@@ -739,76 +690,85 @@ struct CEventFileV1 : public CInterfaceOf<IEventFile>
             switch (attrInformation[attr].type)
             {
             case EATnone:
-                if (!finishAttribute(attr, visitor, mute))
-                    return false;
+                throw makeStringExceptionV(-1, "no data type for attribute %u", attr);
                 break;
             case EATbool:
-                if (!finishAttribute<bool>(attr, visitor, mute))
+                if (!finishAttribute<bool>(attr))
                     return false;
                 break;
             case EATu1:
-                if (!finishAttribute<uint8_t>(attr, visitor, mute))
+                if (!finishAttribute<uint8_t>(attr))
                     return false;
                 break;
             case EATu2:
-                if (!finishAttribute<uint16_t>(attr, visitor, mute))
+                if (!finishAttribute<uint16_t>(attr))
                     return false;
                 break;
             case EATu4:
-                if (!finishAttribute<uint32_t>(attr, visitor, mute))
+                if (!finishAttribute<uint32_t>(attr))
                     return false;
                 break;
             case EATu8:
             case EATtimestamp:
-                if (!finishAttribute<uint64_t>(attr, visitor, mute))
+                if (!finishAttribute<uint64_t>(attr))
                     return false;
                 break;
             case EATstring:
-                if (!finishAttribute(attr, 0, visitor, mute))
+                if (!finishAttribute(attr))
                     return false;
                 break;
             case EATtraceid:
-                if (!finishAttribute(attr, 32, visitor, mute))
+                if (!finishAttribute(attr, 32))
                     return false;
                 break;
             }
         }
         return true;
     }
 
-    bool traverseFooter(IEventVisitor& visitor)
+    bool traverseFooter()
     {
-        (void)visitor.leaveFile(bytesRead);
+        (void)visitor->departFile(bytesRead);
         return true;
     }
 
-    bool finishAttribute(EventAttr attr, IEventVisitor& visitor, bool& mute)
+    bool finishEvent()
     {
         if (mute)
             return true;
-        return reactToVisit(visitor.visitAttribute(attr), mute);
+        mute = false; // reset for next event
```

Comment:
has no effect (it will not clear if it was true in the line above because of the return).